src/handlers.rs

```diff
@@ -0,0 +1,357 @@
+//
+// Copyright (c) 2024 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+use std::marker::PhantomData;
+
+use pyo3::{exceptions::PyAttributeError, prelude::*};
+use zenoh::handlers::{Callback, Dyn, IntoHandler};
+
+use crate::utils::{allow_threads, bail, IntoPython, IntoRust, MapIntoPy, ToPyResult};
+
+#[pyclass]
+#[derive(Clone)]
+pub(crate) struct DefaultHandler;
+
+impl IntoRust for DefaultHandler {
+    type Into = zenoh::handlers::DefaultHandler;
+
+    fn into_rust(self) -> Self::Into {
+        Self::Into::default()
+    }
+}
+
+#[pymethods]
+impl DefaultHandler {
+    #[new]
+    fn new() -> Self {
+        Self
+    }
+}
+
+#[pyclass]
+#[derive(Clone)]
+pub(crate) struct FifoChannel(usize);
+
+impl IntoRust for FifoChannel {
+    type Into = zenoh::handlers::FifoChannel;
+
+    fn into_rust(self) -> Self::Into {
+        Self::Into::new(self.0)
+    }
+}
+
+#[pymethods]
+impl FifoChannel {
+    #[new]
+    fn new(capacity: usize) -> Self {
+        Self(capacity)
+    }
+}
+
+#[pyclass]
+#[derive(Clone)]
+pub(crate) struct RingChannel(usize);
```

Comment:
I'm not sure if naming it as RingChannel is good or not, since Zenoh uses RingBuffer and it's not consistent.

Reply:
In Rust `RingChannel` is being used. It is true that at the beginning of development it was called `RingBuffer` but now it is called `RingChannel`.